pkg/processor/procbuiltin/httprequest.go

```diff
@@ -110,6 +125,10 @@ func httpRequest(
 			// regard status codes over 299 as errors
 			return record.Record{}, cerrors.Errorf("%s: invalid status code %v (body: %q)", processorType, resp.StatusCode, string(body))
 		}
+		// skip if body has no content
+		if resp.StatusCode == http.StatusNoContent {
+			return record.Record{}, processor.ErrSkipRecord
+		}
 
 		r.Payload.After = record.RawData{Raw: body}
```

Comment:
I'm thinking, if it would make sense to move `r.Payload.After` to `r.Payload.Before`? 🤔 Maybe, maybe not, I'm opening the discussion to see what you think 🙂

Reply:
It's definitely an interesting idea... It looks like, if we decide to do it, it should probably be something common to all processors, so maybe we discuss it in details in "Better Processors"?

Reply:
it is interesting and does make sense! 
I agree with Haris about discussing this with the "better processors' work